source/nodejs/tests/ui-tests/src/card-element-utils.ts

```diff
@@ -0,0 +1,347 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { WebElement, By } from "selenium-webdriver";
+import { TestUtils } from "./test-utils";
+import * as Assert from "assert";
+
+export class Element {
+    underlyingElement?: WebElement = undefined;
+
+    protected constructor() {}
+
+    elementWasFound(): boolean
+    {
+        return (this.underlyingElement !== undefined);
+    }
+}
+
+export abstract class Input extends Element {
+    protected constructor() {
+        super();
+    }
+
+    abstract setData(data: string): Promise<void>;
+}
+
+export class InputText extends Input {
+    private constructor(){
+        super();
+    }
+
+    static async getInputWithId(id: string): Promise<InputText>
+    {
+        let input = new InputText();
+        input.underlyingElement = await TestUtils.getInstance().getInput(id, "ac-textInput");
+        return input;
+    }
+
+    override async setData(data: string): Promise<void>
+    {
+        await this.underlyingElement?.click();
+        await this.underlyingElement?.sendKeys(data);
+    }
+
+    async isFocused(): Promise<boolean>
+    {
+        if (this.elementWasFound())
+        {
+            const inputId: string = await this.underlyingElement!.getAttribute("id");
```

Comment:
why do you need await here? what's `!.`. It's the first time I see this operator

Reply:
it's the non-null assertion operator, it helps the compiler as sometimes it doesn't know if the value is null/undefined or not, in this case as I'm calling elementWasFound I know underlyingElement is not null but the compiler will complain